clients/js/src/compat.ts

```diff
@@ -500,20 +499,39 @@ class EnvelopeError extends Error {}
  * Note: MetaMask does not support Web3 methods it doesn't know about, so we have to
  * fall back to manually querying the default gateway.
  */
-async function inferRuntimePublicKeySource(
+async function fetchRuntimePublicKey(
   upstream: UpstreamProvider,
-): Promise<Parameters<typeof fetchRuntimePublicKey>[0]> {
+): Promise<Uint8Array> {
   const isSigner = isEthersSigner(upstream);
+  let chainId: number;
   if (isSigner || isEthersProvider(upstream)) {
-    return {
-      chainId: isSigner
-        ? await upstream.getChainId()
-        : (await upstream.getNetwork()).chainId,
-    };
+    if (
+      'provider' in upstream &&
```

Comment:
This appears to only uses the web3 gateway when the upstream is an ethers signer. The way it was before, it would first check if it's ethers anything, and then pull specifically the provider out because that's all that's needed. Could you ensure that both (non-metamask-backed) ethers provider and signer can use this code path?